src/components/Property/RaiseWorkOrder/SelectPriority.js

```diff
@@ -0,0 +1,63 @@
+import PropTypes from 'prop-types'
+import { Select } from '../../Form'
+import { useState } from 'react'
+import WarningInfoBox from '../../Template/WarningInfoBox'
+
+const SelectPriority = ({
+  priorities,
+  onPrioritySelect,
+  register,
+  errors,
+  priorityCode,
+  priorityCodesWithoutDrs,
+}) => {
+  const [drsScheduled, setDrsScheduled] = useState(false)
+
+  const onSelect = (e) => {
+    setDrsScheduled(priorityCodesWithoutDrs.includes(e.target.value))
+    onPrioritySelect(e)
+  }
+
+  return (
+    <>
+      <Select
+        name="priorityCode"
+        label="Task priority"
+        options={priorities.map((priority) => ({
+          text: priority.description,
+          value: priority.priorityCode.toString(),
+        }))}
+        onChange={onSelect}
+        disabled={true}
+        required={true}
+        register={register({
+          required: 'Please select a priority',
+          validate: (value) =>
+            priorities.map((p) => p.priorityCode).includes(parseInt(value)) ||
+            'Priority is not valid',
+        })}
+        error={errors && errors.priorityCode}
+        widthClass="govuk-!-width-full"
+      />
+      {(drsScheduled || priorityCode == '5') && (
```

Comment:
Added extra layer in case when trade that user selects already has `priorityCode 5 ==> PLANNED`